File: leetcode/0424-longest-repeating-character-replacement/solution.py
class Solution(object):
    def characterReplacement(self, s, k):
        l=0
        res=0
        n=len(s)
        extra={'A':0,'B':0}
        for r in range(n):
            if s[r] in extra.keys():
                extra[s[r]]+=1
            else:
                extra[s[r]]=1
            maxf=max(extra.values())
            if (r-l+1)-maxf>k:
                    extra[s[l]]-=1
                    l+=1
            res=max(res,r-l+1)
        return res
            

        """
        :type s: str
        :type k: int
        :rtype: int
        """
        

File: leetcode/0424-longest-repeating-character-replacement/solution.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l=0;
        int maxlen=0;
        int maxcnt=0;
        vector<int> freq(26,0);
        for(int r=0;r<s.size();r++){
            freq[s[r]-'A']++;
            maxcnt=max(maxcnt,freq[s[r]-'A']);
            while(r-l+1-maxcnt>k){
                freq[s[l]-'A']--;
                l++;
            }
            maxlen=max(maxlen,r-l+1);
        }
        return maxlen;
    }
};
